File: frontend/src/types/checklist.ts
export interface ChecklistItem {
  description: string
  regulatoryReference: string
  timeline: string
  priority: string
  completed?: boolean
}

export interface ChecklistPhase {
  phase: string
  label: string
  items: ChecklistItem[]
}

export interface Checklist {
  tokenClassification: string
  requiredLicenses: string[]
  phases: ChecklistPhase[]
}

File: backend/src/models/checklist.py
"""Models for regulatory checklists."""

from pydantic import BaseModel


class ChecklistItem(BaseModel):
    description: str
    regulatory_reference: str
    timeline: str
    priority: str


class ChecklistPhase(BaseModel):
    phase: str
    label: str
    items: list[ChecklistItem]


class Checklist(BaseModel):
    token_classification: str
    required_licenses: list[str]
    phases: list[ChecklistPhase]
